Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/DriverPlan.hs

```diff
@@ -113,9 +225,23 @@ updateCoinFieldsByDriverId driverId amount = do
 
 instance FromTType' BeamDF.DriverPlan DriverPlan where
   fromTType' BeamDF.DriverPlanT {..} = do
-    pure $
-      Just
-        DriverPlan
+    subscriptionServiceRelatedData <- do
+      case commodityData of
+        Just cd -> do
+          let parsedData = A.fromJSON cd :: A.Result SubscriptionServiceRelatedData
+          case parsedData of
+            A.Success subscriptionServiceRelatedData' -> return subscriptionServiceRelatedData'
+            A.Error _ -> throwError $ InternalError "Error while parsing subscriptionServiceRelatedData"
+        Nothing -> pure NoData
+    case (merchantId, merchantOpCityId) of
+      (Just merchantId', Just merchantOperatingCityId') -> do
+        pure $ Just $ mkDriverPlan (Id merchantId') (Id merchantOperatingCityId') subscriptionServiceRelatedData
+      _ -> do
+        person <- QP.findById (Id driverId) >>= fromMaybeM (PersonNotFound driverId)
```

Comment:
Let it be but also write a backfill query to backfill merchantoperatingCity for all driverPlan.